src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7015,6 +7015,7 @@ private void ApplyMemberPostConditions(BoundExpression? receiverOpt, MethodSymbo
             if (method.MethodKind == MethodKind.LocalFunction
                 && GetTopLevelMethod(method) is { ThisParameter: { } thisParameter })
             {
+                Debug.Assert(receiverSlot == -1);
```

Comment:
Apparently a bunch of tests are failing this assert. Huh. https://dev.azure.com/dnceng-public/public/_build/results?buildId=842224&view=ms.vss-test-web.build-test-results-tab&runId=21841210&resultId=123811&paneView=debug